anymail/backends/sendinblue.py

```diff
@@ -0,0 +1,144 @@
+        # SendinBlue doesn't give any detail on a success
+        # https://developers.sendinblue.com/docs/responses
+
+        return dict()
+
+
+class SendinBluePayload(RequestsPayload):
+
+    def __init__(self, message, defaults, backend, *args, **kwargs):
+        self.all_recipients = []  # used for backend.parse_recipient_status
+
+        http_headers = kwargs.pop('headers', {})
+        http_headers['api-key'] = backend.api_key
+        http_headers['Content-Type'] = 'application/json'
+
+        super(SendinBluePayload, self).__init__(message, defaults, backend, headers=http_headers, *args, **kwargs)
+
+    def get_api_endpoint(self):
+        return "smtp/email"
+
+    def init_payload(self):
+        self.data = {  # becomes json
+            'to': [],
+            'cc': [],
+            'bcc': [],
+            'sender': [],
+        }
+
+    def serialize_data(self):
+        """Performs any necessary serialization on self.data, and returns the result."""
+
+        headers = self.data["headers"]
+        if "Reply-To" in headers:
+            # Reply-To must be in its own param
+            reply_to = headers.pop('Reply-To')
+            self.set_reply_to(parse_address_list([reply_to]))
+        if len(headers) > 0:
+            self.data["headers"] = dict(headers)  # flatten to normal dict for json serialization
+        else:
+            del self.data["headers"]  # don't send empty headers
+
+        return self.serialize_json(self.data)
+
+    #
+    # Payload construction
+    #
+
+    @staticmethod
+    def email_object(email):
+        """Converts EmailAddress to SendinBlue API array"""
+        email_object = dict()
+        email_object['email'] = email.addr_spec
+        if email.display_name:
+            email_object['name'] = email.display_name
+        return email_object
+
+    def set_from_email(self, email):
+        self.data["sender"] = self.email_object(email)
+
+    def set_recipients(self, recipient_type, emails):
+        assert recipient_type in ["to", "cc", "bcc"]
+        if emails:
+            # Normally, exactly one "personalizations" entry for all recipients
+            # (Exception: with merge_data; will be burst apart later.)
+            self.data[recipient_type] = [self.email_object(email) for email in emails]
+            self.all_recipients += emails  # used for backend.parse_recipient_status
+
+    def set_subject(self, subject):
+        if subject != "":  # see note in set_text_body about template rendering
+            self.data["subject"] = subject
+
+    def set_reply_to(self, emails):
+        # SendinBlue only supports a single address in the reply_to API param.
+        if len(emails) > 1:
+            self.unsupported_feature("multiple reply_to addresses")
+        if len(emails) > 0:
+            self.data['replyTo'] = self.email_object(emails[0])
+
+    def set_extra_headers(self, headers):
+        self.data['headers'] = [header for header in headers]
```

Comment:
SiB expects an "object" here, but doesn't say much more. I'm guessing it's a mapping of header field-name to value, which is what the `headers` param already is. So you can probably just do `self.data['headers'] = headers`.

Reply:
The documentation of v2 have more detail on the `headers` parameter :

|Input Parameter | Description|
| -- | -- |
| headers | The headers will be sent along with the mail  headers in original email. Example: array(“Content-Type”=>”text/html;  charset=iso-8859-1″). You can use commas to separate multiple headers  [Optional]. |

I guess it's just the same thing.